src/transformers/integrations/bitsandbytes.py

```diff
@@ -74,13 +76,14 @@ class `Int8Params` from `bitsandbytes`.
                 new_value = old_value.to(device)
             elif isinstance(value, torch.Tensor):
                 new_value = value.to("cpu")
-                if value.dtype == torch.int8:
+                if value.dtype in (torch.int8, torch.uint8):
                     is_8bit_serializable = version.parse(importlib.metadata.version("bitsandbytes")) > version.parse(
                         "0.37.2"
                     )
                     if not is_8bit_serializable:
+                        bits_ = 4 if is_4bit else 8
                         raise ValueError(
-                            "Detected int8 weights but the version of bitsandbytes is not compatible with int8 serialization. "
+                            f"Detected {bits_} weights but the version of bitsandbytes is not compatible with {bits_} bit serialization. "
                             "Make sure to download the latest `bitsandbytes` version. `pip install --upgrade bitsandbytes`."
                         )
```

Comment:
Then here we would break it down in two cases -> 8bit serializable through `is_8bit_serializable` and 4-bit serializable through a new bool `is_4bit_serializable`

Reply:
Done. Separated 2 cases more clearly.